contracts/components/staking/stakeSubjectHandling/IStakeSubjectHandler.sol

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: UNLICENSED
+// See Forta Network License: https://github.com/forta-network/forta-contracts/blob/master/LICENSE.md
+
+pragma solidity ^0.8.9;
+
+import "./IStakeSubject.sol";
+
+interface IStakeSubjectHandler {
+    event StakeSubjectChanged(address newHandler, address oldHandler);
+    function setStakeSubject(uint8 subjectType, address subject) external;
+    function getStakeSubject(uint8 subjectType) external view returns (address);
+    function activeStakeFor(uint8 subjectType, uint256 subject) external view returns (uint256);
+    function maxStakeFor(uint8 subjectType, uint256 subject) external view returns (uint256);
```

Comment:
The name `subject` is used for `address subject` in `setStakeSubjet` and also for `uint subject` in `maxStakeFor` (etc.) but I believe they refer to completely different things.

In general I think "subject" and related concepts are quite overloaded, and it makes reading the code difficult.

In particular, note that in this PR there was this series of renames:
- `IStakeController` -> `IStakeSubjectHandler`
- `setStakeSubjectHandler` -> `setStakeSubject`

The concept of "stake subject handler" was reused for the interface name, but for almost the opposite thing. Note that the parameters of the event `StakeSubjectChanged` are still using the "handler" term!

---

The way I understand all this:
- A "subject" is an individual node runner, scanner, or agent.
  - Each of these individuals is a "stake subject" in the sense that they can have stake put on them.
- Each individual has a "subject id" that identifies them, which are respectively an integer (sequential), an address, and an integer.
  - What is called `subject` in `maxStakeFor` and all the other getters could be called `subjectId`, but this distinction is not so important (because a subject is always represented by its id).
- There are "subject registries", respectively NodeRunnerRegistry, ScannerRegistry (being phased out), and AgentRegistry. The subject registries contain various data about the subjects themselves, and in particular about their stake thresholds and allocations (although the stake itself is elsewhere, in the FortaStaking contract).
  - **Suggestion:** `setStakeSubject(..., address subject)` should be called or `setStakeSubjectRegistry(..., address registry)` (or something involving "registry").
- There is a central contract that answers queries about stake and stake-related data, by directing them to the appropriate registry.
  - This is currently the so-called "stake subject handler", but as noted above this term seemed to refer to the registries previously.
  - **Suggestion:** `StakeSubjectHandler` needs a different name. IMO even `StakeController` is better, but I can also suggest `StakeSubjectQueries`. "Handler" is not right, this contract is not handling, it is directing queries to the appropriate registry.

Reply:
Apologies for the inconsistencies, I might have left some stray "handlers".

Minor correction: IStakeController -> IStakeSubjectHandle should be FortaStakingParameters -> IStakeSubjectHandler

> Each individual has a "subject id" that identifies them, which are respectively an integer (sequential), an address, and an integer.

It's all integers. In the case of ScannerSubject it is still an integer, converted from address

> What is called subject in maxStakeFor and all the other getters could be called subjectId, but this distinction is not so important (because a subject is always represented by its id).

unification might be important, but not sure if I have time to go throught that one.

> Suggestion: setStakeSubject(..., address subject) should be called or setStakeSubjectRegistry(..., address registry) (or something involving "registry").

Why not

> Suggestion: StakeSubjectHandler needs a different name. IMO even StakeController is better, but I can also suggest StakeSubjectQueries. "Handler" is not right, this contract is not handling, it is directing queries to the appropriate registry.

Controller is very generic too, since it is not controlling anything. `StakeSubjectRepository` sounds better IMO, what do you think?

Reply:
StakeSubjectGateway it is